src/test/java/org/mule/extensions/jms/test/transaction/JmsTransactionalListenerTestCase.java

```diff
@@ -0,0 +1,147 @@
+import io.qameta.allure.Step;
+import io.qameta.allure.Story;
+
+@Feature(JMS_EXTENSION)
+@Story(TRANSACTION)
+public class JmsTransactionalListenerTestCase extends JmsAbstractTestCase {
+
+  @Rule
+  public SystemProperty initialDestination =
+      new SystemPropertyLambda("initialDestination", () -> newDestination("initialDestination"));
+
+  @Rule
+  public SystemProperty finalDestination = new SystemPropertyLambda("finalDestination", () -> newDestination("finalDestination"));
+
+  @Rule
+  public SystemProperty consumeDestination =
+      new SystemPropertyLambda("consumeDestination", () -> newDestination("consumeDestination"));
+
+  private String message;
+
+  @Override
+  protected String[] getConfigFiles() {
+    return new String[] {"transactions/jms-transactional-listener.xml", "config/activemq/activemq-default.xml"};
+  }
+
+  @Test
+  @Description("Verifies that transactions started by a listener work well when next jms operation is set to join if" +
+      " possible")
+  public void txListenerWithDefaultTxActionOnNextOperation() throws Exception {
+    message = buildMessage(Actions.NOTHING);
+    publishMessage(message, initialDestination.getValue());
+    publishMessage(message, consumeDestination.getValue());
+
+    ((Flow) getFlowConstruct("txListenerWithDefaultTxActionOnNextOperation")).start();
+
+    assertMessageOnDestination(message, finalDestination.getValue());
+    assertEmptyDestination(initialDestination.getValue());
+  }
+
+  @Test
+  @Description("Verifies that rollbacks work as expected when transactions are started by a listener and next jms " +
+      "operation is set to join if possible")
+  public void txListenerWithDefaultTxActionOnNextOperationRolledBack() throws Exception {
+    message = buildMessage(Actions.EXPLODE);
+    publishMessage(message, initialDestination.getValue());
+    publishMessage(message, consumeDestination.getValue());
+
+    ((Flow) getFlowConstruct("txListenerWithDefaultTxActionOnNextOperation")).start();
+
+    assertEmptyDestination(finalDestination.getValue());
+    assertMessageOnDestination(message, initialDestination.getValue());
+  }
+
+  @Test
+  @Description("Verifies that transactions started by listener work well even though next jms operation is set to not" +
+      " join them")
+  public void txListenerWithNotSupportedTxActionOnNextOperation() throws Exception {
+    message = buildMessage(Actions.NOTHING);
+    publishMessage(message, initialDestination.getValue());
+    publishMessage(message, consumeDestination.getValue());
+
+    ((Flow) getFlowConstruct("txListenerWithNotSupportedTxActionOnNextOperation")).start();
+
+    assertMessageOnDestination(message, finalDestination.getValue());
+    assertEmptyDestination(initialDestination.getValue());
+  }
+
+  @Test
+  @Description("Verifies that rollback of transactions started by a listener does not involve next jms operation" +
+      " if set not to join them")
+  @Ignore("MULE-13711")
```

Comment:
Add the Issue annotation

Reply:
Indeed I had to have removed this because the issue was already fixed